File: python/meteroid/models/customer_event_data.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .customer_id import CustomerId


@dataclasses.dataclass
class CustomerEventData(BaseModel):
    currency: str

    custom_properties: t.Dict[str, t.Any]
    """User-defined custom property values, keyed by definition key."""

    customer_id: CustomerId

    invoicing_emails: t.List[str]

    name: str

    alias: t.Optional[str] = None

    billing_email: t.Optional[str] = None

    phone: t.Optional[str] = None

File: go/customer_event_data.go
// this file is @generated
package meteroid

type CustomerEventData struct {
	Alias *string `json:"alias,omitempty"`

	BillingEmail *string `json:"billing_email,omitempty"`

	Currency string `json:"currency"`

	// User-defined custom property values, keyed by definition key.
	CustomProperties RequiredMap[any] `json:"custom_properties"`

	CustomerId CustomerId `json:"customer_id"`

	InvoicingEmails RequiredSlice[string] `json:"invoicing_emails"`

	Name string `json:"name"`

	Phone *string `json:"phone,omitempty"`
}
